backend/src/controllers/OrderController.ts

```diff
@@ -0,0 +1,122 @@
+import { Request, Response } from 'express';
+import Stripe from 'stripe';
+import Restaurant, { MenuItemType } from '../models/restaurant';
+import Order from '../models/order';
+
+const STRIPE = new Stripe(process.env.STRIPE_API_KEY as string);
+const FRONTEND_URL = process.env.FRONTEND_URL as string;
+
+type CheckoutSessionRequest = {
+  cartItems: {
+    menuItemId: string;
+    name: string;
+    quantity: number;
+  }[];
+  deliveryDetails: {
+    email: string;
+    name: string;
+    addressLine1: string;
+    city: string;
+  };
+  restaurantId: string;
+};
+
+const createCheckoutSession = async (req: Request, res: Response) => {
+  try {
+    const checkoutSessionRequest: CheckoutSessionRequest = req.body as CheckoutSessionRequest;
+    const restaurant = await Restaurant.findById(checkoutSessionRequest.restaurantId);
+    if (!restaurant) {
+      return res.status(404).send('Restaurant not found');
+    }
+
+    const newOrder = new Order({
+      restaurant: restaurant._id,
+      user: req.userId,
+      status: 'placed',
+      cartItems: checkoutSessionRequest.cartItems,
+      deliveryDetails: checkoutSessionRequest.deliveryDetails,
+      createdAt: new Date(),
+    });
+
+    const lineItems = await createLineItems(checkoutSessionRequest, restaurant.menuItems);
+    const session = await createSession(
+      lineItems,
+      newOrder._id.toString(),
+      restaurant.deliveryPrice,
+      restaurant._id.toString(),
+    );
+
+    if (!session.url) {
+      return res.status(500).send('Error creating checkout session');
+    }
+
+    await newOrder.save();
+    return res.status(200).json({ url: session.url });
+  } catch (error) {
+    console.log(error);
+    res.status(500).send('Error creating checkout session');
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/46#issuecomment-2423673837>#b6ffb4</a><span> - 10/19/2024, 08:13 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The error handling in the createCheckoutSession function is insufficient. It logs the error to the console, which may expose sensitive information in production logs, and returns a generic error message to the client.
</div>
<div id="fix">
<b>Fix:</b> Consider implementing more robust error handling by using console.error for logging and providing more informative (but safe) error messages to the client. Distinguish between Stripe-specific errors and other unexpected errors to provide appropriate responses.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -55,4 +55,11 @@
 -  } catch (error) {
 -    console.log(error);
 -    res.status(500).send('Error creating checkout session');
 -  }
 +  } catch (error) {
 +    console.error('Error in createCheckoutSession:', error);
 +    if (error instanceof Stripe.StripeError) {
 +      res.status(400).json({ message: 'Payment processing error. Please try again.' });
 +    } else {
 +      res.status(500).json({ message: 'An unexpected error occurred. Please try again later.' });
 +    }
 +  }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged